### Домашняя работа №3

##### Задание №1

1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np
from math import factorial as facto

In [2]:
# list1 - список значений зарплат

list1 = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
list1 = np.array(sorted(list1))
list1

array([ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150])

In [3]:
# mean_1ist1 - среднее арифметическое

mean_list1 = sum(list1)/list1.size
mean_list1

65.3

In [4]:
# d - дисперсия
# dispersion(_non) - смещённая(несмещённая) дисперсия

d = []

for i in list1:
    sqrt_err = (i - mean_list1)**2
    d.append(sqrt_err)

dispersion = sum(d) / list1.size                   # смещённая дисперсия
dispersion_non = sum(d) / (list1.size - 1)         # несмещённая дисперсия
dispersion, dispersion_non

(950.11, 1000.1157894736842)

In [5]:
# sigma(_non) - среднее квадратичное отклонение

sigma = np.sqrt(dispersion)
sigma_non = np.sqrt(dispersion_non)
sigma, sigma_non

(30.823854398825596, 31.624607341019814)

Ответ: по итогам расчётов мы получили следующие результаты:

    65.3 - среднее арифметическое (mean_1ist1); 
    30.8, 31.6 - среднее квадратичное отклонение (sigma, sigma_non); 
    950.1 - смещенную... (dispersion);
    1000.1 - и несмещенную оценки дисперсий для данной выборки (dispersion_non).

##### Задание №2

2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. 
Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. 
Какова вероятность того, что 3 мяча белые?

In [6]:
def binom(n, k):
    return facto(n) / (facto(k) * facto(n - k))

In [7]:
# (P1) - вероятность 2 бел. из 2 в 1-м ящике (P21) * на 1 бел. из 4 во 2-м ящике (P12):

comb_1 = binom(8, 2)                                 
P21 = binom(5, 2) / comb_1

comb_2 = binom(12, 4)                                
P12 = binom(5, 1) * binom(7, 3) / comb_2

P1 = P21 * P12
P1

0.12626262626262627

In [8]:
# (P2) - вероятность 0 бел. из 2 в 1-м ящике (P01) * на 3 бел. из 4 во 2-м ящике (P32):

P01 = binom(5, 0) * binom(3, 2) / comb_1

P32 = binom(5, 3) * binom(7, 1) / comb_2

P2 = P01 * P32
P2

0.01515151515151515

In [9]:
# (P3) - вероятность 1 бел. из 2 в 1-м ящике (P11) * на 2 бел. из 4 во 2-м ящике (P22):

P11 = binom(5, 1) * binom(3, 1) / comb_1

P22 = binom(5, 2) * binom(7, 2) / comb_2

P3 = P11 * P22
P3

0.22727272727272727

In [10]:
P = P1 + P2 + P3
P

0.3686868686868687

Ответ: вероятность того, что 3 мяча белые Р = 0.36868

##### Задание №3

3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [11]:
PA = 0.9
PB = 0.8
PC = 0.6

In [12]:
P = PA + PB + PC
P

2.3000000000000003

In [13]:
PA_shoot = PA / P
PB_shoot = PB / P
PC_shoot = PC / P
PA_shoot, PB_shoot, PC_shoot

(0.3913043478260869, 0.34782608695652173, 0.26086956521739124)

Ответ:  вероятность того, что выстрел произведен: 
        
    0.39130 - первым спортсменом (PA_shoot);
    0.34782 - вторым спортсменом (PB_shoot);
    0.26086 - третьим спортсменом (PC_shoot).

##### Задание №4

4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [14]:
PA = 0.8
PB = 0.7
PC = 0.9

In [15]:
P = PA + PB + 2 * PC
P

3.3

In [16]:
P_fa = PA / P
P_fb = PB / P
P_fc = 2 * PC / P
P_fa, P_fb, P_fc

(0.24242424242424246, 0.21212121212121213, 0.5454545454545455)

Ответ: вероятность, что он учится: 
        
    0.24242 - на факультете A (P_fa);
    0.21212 - на факультете B (P_fb);
    0.54545 - на факультете C (P_fc).

#### Задание №5

5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, 
для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали 
в). хотя бы одна деталь г). от одной до двух деталей?

In [17]:
P1 = 0.1
P2 = 0.2
P3 = 0.25

In [18]:
P = P1 * P2 * P3
P

0.005000000000000001

In [19]:
detail_2 = P1 * P *(1-P3) + P1 * P3 * (1-P2) + P3 * P2 * (1-P1)
detail_1 = 1 - ((1-P1) * (1-P2) * (1-P3))
detail_1_2 = detail_2 + (P1) * (1-P2) * (1-P3) + (P2) * (1-P1) * (1-P2) + (P3) * (1-P2) * (1-P1)
detail_2, detail_1, detail_1_2

(0.06537500000000002, 0.45999999999999996, 0.4493750000000001)

Ответ: вероятность того, что в первый месяц выйдут из строя: 
        
    0.005 - все детали (P);
    0.065 - только две детали (detail_2);
    0.459 - хотя бы одна деталь (detail_1);
    0.449 - от одной до двух деталей (detail_1_2).